### DISCUSSION TITLE
[질문] Binding Adapter 질문

### DISCUSSION THREAD

**Discussion by boogi-woogi** [TEAM-DAME/Dame-Android#106] [2. Q&A] [answered]
![image](https://user-images.githubusercontent.com/88770426/204081908-cf5b4d3d-ee84-4b12-b90b-f7ff4045f489.png)
core-ui에 정의되어 있는 binding adapter에 필요한 것들을 추가로 구현하여 사용하려고 했습니다.
하지만 feature/notification 패키지 안에 있는 notifiaction을 위한 data 클래스들을 인식하지 못해서 아래의 사진과 같이 notification package안에 따로  binding adpater를 만들어서 사용하려고 하는데 이렇게 해도 되는 것인지 궁금합니다!
![image](https://user-images.githubusercontent.com/88770426/204081954-ad573404-5b5b-4a60-afde-aa4913474cdc.png)

**Comment by onseok** [comment, marked as answer]
사진의 내용을 보니 알림 타입에 따라 텍스트뷰에 다른 string 값들을 불러오는 것 같은데, 해당 비즈니스 로직은 뷰모델에서 구현하고, 뷰모델의 livedata를 해당 TextView의 text 속성에 databinding 하는 방식이 더 좋아보입니다.
(모든 로직을 바인딩어댑터에서 처리한다면 뷰에 보여줄 데이터를 가공하고 관리하는 역할을 하는 뷰모델이 서러워질 것 같습니다...)

Binding Adapter는 여러 간단한 커스텀 세터들을 설정하기 위해 쓰인다고 합니다.
[참고 링크](https://stackoverflow.com/questions/56305560/what-is-the-use-of-binding-adapter-in-android)

**Reply by boogi-woogi** [reply]
수정하겠습니다! 감사합니다~~
